src/Bicep.Core.Samples/InvalidExpressions_LF/main.diagnostics.bicep

```diff
@@ -323,6 +323,7 @@ var badInnerArrayIndexer = sampleObject.myInner.otherArr['s']
 var badIndexer = sampleObject.myStr['s']
 //@[17:35) [BCP076 (Error)] Cannot index over expression of type 's'. Arrays or objects are required. |sampleObject.myStr|
 var badInnerArray = sampleObject.myInner.fakeArr['s']
+//@[20:48) [BCP076 (Error)] Cannot index over expression of type error. Arrays or objects are required. |sampleObject.myInner.fakeArr|
```

Comment:
>Cannot [](start = 28, length = 6)

We might have a bug in the indexer logic. The error below is the root cause and this error shouldn't be emitted.

Reply:
(I think in general a diagnostic that mentions the "error" type is suspicious.)

---
In reply to: [497794858](https://github.com/Azure/bicep/pull/578#discussion_r497794858) [](ancestors = 497794858)

Reply:
Great catch. The checks in type validation were just checking for `errors.Any()` whereas it's possible to have an errortype with no diagnostics attached. I've updated all these checks to also look for any sub-expressions of type 'error', which fixes this.